File: client/android/app/src/main/java/org/cna/keyple/demo/sale/android/nfc/slave/data/model/CardReaderResponse.java
/********************************************************************************
 * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
 ********************************************************************************/
package org.cna.keyple.demo.sale.android.nfc.slave.data.model;

public class CardReaderResponse {

    public final Status status;

    public final Integer ticketsNumber;

    public final String cardType, lastValidation, seasonPassExpiryDate;

    public CardReaderResponse(Status status, Integer ticketsNumber, String cardType,
            String lastValidation, String seasonPassExpiryDate) {
        this.status = status;
        this.ticketsNumber = ticketsNumber;
        this.cardType = cardType;
        this.lastValidation = lastValidation;
        this.seasonPassExpiryDate = seasonPassExpiryDate;
    }
}

File: client/android/app/src/main/kotlin/org/eclipse/keyple/demo/remote/data/model/CardReaderResponse.kt
/*
 * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
 *
 * See the NOTICE file(s) distributed with this work for additional information
 * regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License 2.0 which is available at http://www.eclipse.org/legal/epl-2.0
 *
 * SPDX-License-Identifier: EPL-2.0
 */
package org.eclipse.keyple.demo.remote.data.model

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class CardReaderResponse(
    val status: Status,
    val cardType: String,
    val ticketsNumber: Int,
    val titlesList: ArrayList<CardTitle>,
    val lastValidationsList: ArrayList<Validation>,
    val seasonPassExpiryDate: String
): Parcelable